modules/order/src/OrderStorage.php

```diff
@@ -83,10 +85,33 @@ public function loadUnchanged($id) {
   /**
    * {@inheritdoc}
    */
+  protected function doPreSave(EntityInterface $entity) {
+    $id = parent::doPreSave($entity);
+    /** @var \Drupal\commerce_order\Entity\OrderInterface $entity */
+    if ($entity->getRefreshState() == OrderInterface::REFRESH_ON_SAVE) {
+      $this->orderRefresh->refresh($entity);
+    }
+    // Only the REFRESH_ON_LOAD state needs to be persisted on the entity.
+    if ($entity->getRefreshState() != OrderInterface::REFRESH_ON_LOAD) {
```

Comment:
To be honest it took me a bit to grok this.

I would if 

```
switch (refreshState) :

  case REFRESH_ON_LOAD:
     // We do nothing so this persists.
  case REFRESH_ON_SAVE:
    refresh($entity);
   // no break so it also uses skip/default

  case REFRESH_SKIP:
  default:
    setRefreshState(NULL);
  break;

endswitch
```

Little bit easier to understand.